File: example/android/app/src/main/java/com/example/intercomreactnative/MainNotificationService.java
package com.example.intercomreactnative;

import com.google.firebase.messaging.FirebaseMessagingService;
import com.google.firebase.messaging.RemoteMessage;
import com.intercomreactnative.IntercomModule;

public class MainNotificationService extends FirebaseMessagingService {

  @Override
  public void onNewToken(String refreshedToken) {
    IntercomModule.sendTokenToIntercom(getApplication(), refreshedToken);
    //DO LOGIC HERE
  }

  public void onMessageReceived(RemoteMessage remoteMessage) {
    if (IntercomModule.isIntercomPush(remoteMessage)) {
      IntercomModule.handleRemotePushMessage(getApplication(), remoteMessage);
    } else {
      //HANDLE NOT INTERCOM MESSAGE
    }
  }
}

File: examples/with-notifications/android/app/src/main/java/com/example/withnotifications/MainNotificationService.kt
package com.example.withnotifications

import com.google.firebase.messaging.FirebaseMessagingService
import com.google.firebase.messaging.RemoteMessage
import com.intercom.reactnative.IntercomModule

class MainNotificationService : FirebaseMessagingService() {

	override fun onNewToken(refreshedToken: String) {
	    IntercomModule.sendTokenToIntercom(application, refreshedToken)
	    // DO HOST LOGIC HERE
	}

	override fun onMessageReceived(remoteMessage: RemoteMessage) {
	    if (IntercomModule.isIntercomPush(remoteMessage)) {
	     	IntercomModule.handleRemotePushMessage(application, remoteMessage)
	    } else {
	        // DO HOST LOGIC HERE
	    }
	}
}
